src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -280,6 +299,47 @@ private void validateExistingIndex(IndexMetaData currentIndexMetaData, IndexMeta
                     }
                 }
 
+                private IndexMetaData updateIndexSettings(IndexMetaData indexMetaData, Settings changeSettings, String[] ignoreSettings) {
+                    if (changeSettings.names().isEmpty() && ignoreSettings.length == 0) {
+                        return indexMetaData;
+                    }
+                    IndexMetaData.Builder builder = IndexMetaData.builder(indexMetaData);
+                    Map<String, String> settingsMap = newHashMap(indexMetaData.settings().getAsMap());
+                    List<String> simpleMatchPatterns = newArrayList();
+                    for (String ignoredSetting : ignoreSettings) {
+                        if (!Regex.isSimpleMatchPattern(ignoredSetting)) {
+                            if (!UNREMOVABLE_SETTINGS.contains(ignoredSetting)) {
```

Comment:
Can you invert the if statement to remove the `!` and use `== false`? I think it makes things much easier to miss and reduces the chance of typos for future changes